lib/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart

```diff
@@ -0,0 +1,50 @@
+part of 'ban_name_rule.dart';
+
+class _Visitor extends RecursiveAstVisitor<void> {
+  final Map<String, _BanNameConfigEntry> _entryMap;
+
+  final _nodes = <_NodeWithMessage>[];
+
+  Iterable<_NodeWithMessage> get nodes => _nodes;
+
+  _Visitor(List<_BanNameConfigEntry> entries)
+      : _entryMap = Map.fromEntries(entries.map((e) => MapEntry(e.ident, e)));
+
+  @override
+  void visitSimpleIdentifier(SimpleIdentifier node) {
+    super.visitSimpleIdentifier(node);
+    _visitIdent(node, node);
+  }
+
+  @override
+  void visitPrefixedIdentifier(PrefixedIdentifier node) {
+    super.visitPrefixedIdentifier(node);
+    _visitIdent(node, node.identifier);
+    _visitIdent(node, node.prefix);
+  }
+
+  @override
+  void visitLibraryIdentifier(LibraryIdentifier node) {
+    super.visitLibraryIdentifier(node);
+    for (final component in node.components) {
+      _visitIdent(node, component);
+    }
+  }
+
+  void _visitIdent(Expression node, SimpleIdentifier ident) {
```

Comment:
Since it's an identifier, that means it will trigger for variable, method, function, class names, etc. (everything, that has an identifier). Have you thought about edge case when you can have a variable and a method called the same way (somehow) and triggering on one of them will be considered as a false-positive? Maybe there should be some config to limit the type of what contains that identifier?

Reply:
(discussed below)